packages/devtools/lib/src/memory/memory_controller.dart

```diff
@@ -76,7 +76,7 @@ class MemoryController {
     final Map resetArg = reset ? {'reset': 'true'} : {};
 
     final Response response = await serviceManager.service.callMethod(
-      '_getAllocationProfile',
```

Comment:
Devtools needs to be able to work with the latest stable version of Flutter. Having the change land in tip of trunk Flutter is not enough.
I would suggest one of the follow approaches:
A) Choose to use _getAllocationProfile or getAllocationProfile based on the VM Service version.
B) Capability detection based on whether the service says it exposes 'getAllocationProfile'. Not sure if the VMService exposes an api indicating what methods are available. I know it does exist for the service extension case which is waht we do in that case.